public/app/features/alerting/unified/components/rules/EditRuleGroupModal.tsx

```diff
@@ -315,44 +287,55 @@ export function EditCloudGroupModal(props: ModalProps): React.ReactElement {
     notifyApp.error('There are errors in the form. Correct the errors and retry.');
   };
 
-  const rulerRuleRequests = useUnifiedAlertingSelector((state) => state.rulerRules);
-  const groupfoldersForSource = rulerRuleRequests[sourceName];
-
-  const groupWithRules = getGroupFromRuler(groupfoldersForSource?.result, groupName, nameSpaceName);
-  const rulesWithoutRecordingRules: RulerRuleDTO[] =
-    groupWithRules?.rules.filter((rule: RulerRuleDTO) => !isRecordingRulerRule(rule)) ?? [];
+  const rulesWithoutRecordingRules = compact(
+    group.rules.map((r) => r.rulerRule).filter((rule) => !isRecordingRulerRule(rule))
+  );
   const hasSomeNoRecordingRules = rulesWithoutRecordingRules.length > 0;
+  const modalTitle =
+    intervalEditOnly || isGrafanaManagedGroup ? 'Edit evaluation group' : 'Edit namespace or evaluation group';
 
   return (
-    <Modal
-      className={styles.modal}
-      isOpen={true}
-      title={folderAndGroupReadOnly ? 'Edit evaluation group' : `Edit ${nameSpaceLabel} or evaluation group`}
-      onDismiss={onClose}
-      onClickBackdrop={onClose}
-    >
+    <Modal className={styles.modal} isOpen={true} title={modalTitle} onDismiss={onClose} onClickBackdrop={onClose}>
       <FormProvider {...formAPI}>
         <form onSubmit={(e) => e.preventDefault()} key={JSON.stringify(defaultValues)}>
           <>
             <Field
               label={
-                <Label htmlFor="namespaceName">
+                <Label
+                  htmlFor="namespaceName"
+                  description={
+                    !isGrafanaManagedGroup &&
+                    'Change the current namespace name. Moving groups between namespaces is not supported'
+                  }
+                >
                   <Stack gap={0.5}>
                     {nameSpaceLabel}
-                    <InfoIcon text={spaceNameInfoIconLabel} />
+                    {!isGrafanaManagedGroup && <InfoIcon text={spaceNameInfoIconLabel} />}
                   </Stack>
                 </Label>
               }
               invalid={!!errors.namespaceName}
               error={errors.namespaceName?.message}
             >
-              <Input
-                id="namespaceName"
-                readOnly={folderAndGroupReadOnly}
-                {...register('namespaceName', {
-                  required: 'Namespace name is required.',
-                })}
-              />
+              <Stack gap={1} direction="row">
+                <Input
+                  id="namespaceName"
+                  readOnly={intervalEditOnly || isGrafanaManagedGroup}
+                  {...register('namespaceName', {
+                    required: 'Namespace name is required.',
+                  })}
+                  className={styles.formInput}
+                />
+                {isGrafanaManagedGroup && (
+                  <LinkButton
+                    href={props.folderUrl}
```

Comment:
Should we render this component if `folderUrl` is falsey? I think it doesn't make a whole lot of sense to show this button if clicking it doesn't take you anywhere :)

Reply:
We shouldn't! Good catch